cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/cmd"
+	"github.com/vmware-tanzu/tanzu-framework/util/carvelhelpers"
+	"github.com/vmware-tanzu/tanzu-framework/util/kube"
+)
+
+const dotCharacter = 46
+
+func downloadClusterEssentialBundle(clusterEssentialBundlePath string) error {
+	var outputBuf, errorBuf bytes.Buffer
+	writerUI := ui.NewWriterUI(&outputBuf, &errorBuf, nil)
+	outputDir := "tmp"
```

Comment:
This is being used in multiple places. Avoid that and use a constant so that any change to this can be done at a single place.